doc/BUILD.rst

```diff
@@ -0,0 +1,730 @@
+the platform, as it contains headers and tools for building native macOS apps.
+
+.. code:: sh
+
+   xcode-select --install
+
+There are additional command line tools that you'll need. You can acquire those
+by using `Homebrew <https://brew.sh/>`__.
+
+Follow the instructions for installing Homebrew, and then install the following
+tools:
+
+.. code:: sh
+
+   brew install meson cmake pkg-config docutils pygments
+
+You can then install dependencies from the *Optional Dependencies* list.
+
+As of 2021-08-05, you should **not** install the following packages via
+Homebrew, as the versions available do not compile against Taisei correctly.
+If you're having mysterious errors, ensure that they're not installed.
+
+-  ``spirv-tools``
+-  ``spirv-cross``
+
+.. code:: sh
+
+   brew remove spirv-tools spirv-cross
+
+In addition, if you're trying to compile on an older version of macOS
+(i.e: <10.12), SDL2 may not compile correctly on Homebrew (as of 2019-02-19).
+Let ``meson`` pull in the corrected version for you via subprojects.
+
+You can also install `create-dmg <https://github.com/create-dmg/create-dmg>`__
+for packaging ``.dmg`` files, which enables some additional options such as
+positioning of icons in the ``.dmg``.
+
+Windows
+'''''''
+
+Taisei uses `mstorsjo/llvm-mingw <https://github.com/mstorsjo/llvm-mingw>`__ to
+achieve cross-compiling for Windows. Microsoft's native C compiler toolchain
+simply does not support the things Taisei needs to compile correctly, including
+fundamental things like
+`complex numbers <https://en.wikipedia.org/wiki/Complex_number>`__.
+
+
+You can use ``llvm-mingw`` too, or you can check if your distro has any
+``mingw64`` cross-compiler toolchains available as well. That's is just the one
+that works for us.
+
+Additionally, you can install `nsis <https://nsis.sourceforge.io/Main_Page>`__
+(>= 3.0) for packaging Windows installer ``.exe`` files. (However, you can
+still package ``.zip`` files for Windows without it.)
+
+On macOS, you're probably better off using Docker and the
+`Docker container <https://hub.docker.com/r/mstorsjo/llvm-mingw/>`__ that
+``llvm-mingw`` provides, and installing ``nsis`` on top of it.
+
+Another options for Windows-based computers is leveraging Windows
+10's
+`Windows For Linux (WSL) Subsystem <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`__
+to cross-compile to Windows using their Ubuntu image. You can also potentially
+use a ``mingw64`` toolchain directly on Windows, however that isn't supported
+or recommended, as it's generally more trouble than its worth.
+
+Checking Out Code
+-----------------
+
+First, you'll need to checkout the repository. You can do that with the
+following:
+
+.. code:: sh
+
+   git clone --recurse-submodules https://github.com/taisei-project/taisei
+
+The ``git submodule update --init --recursive`` line is absolutely necessary,
+or Taisei will not build, as it will be missing many of the dependencies its
+needs to compile.
```

Comment:
Seems like you forgot to update this paragraph, since that line doesn't appear in the instructions anymore. It could be rewritten like this:

> The `--recurse-submodules` parameter is necessary. If you accidentally omitted it, you can run `cd taisei && git submodule update --init --recursive` to fix that.